### PR TITLE
[RLlib] Implement Pi head for Policy-Gradient-Based Methods

### PR DESCRIPTION
**Pull request by smorad**
## Why are these changes needed?

This produces the Pi's used by the new `RLModule` paradigm. `TorchPi` will generate actions for policy-gradient based methods, such as PG, PPO, A2C, A3C, and IMPALA.

TL;DR:  See my beautiful diagram I worked really hard on

rllib-actions (2)

The `configs` are fairly straightforward. We provide a `PolicyGradientPiConfig` with a `DecomposeSpace` method that maps action spaces to `DistributionConfigs`s. Call `DistributionConfig.build(logits)` to construct the distribution at runtime.  Users wishing to use their own custom distributions should implement their own `MyCustomPiConfig` and override `DecomposeSpace`. This can be done via configs, e.g. `PPOConfig(pi=MyCustomPiConfig)`.

`TorchPi` calls `DistributionConfig.build(logits)` during each forward pass. For ease-of-use, `TorchPi` outputs a `MultiTorchDistribution` instance that wraps a collection of `TorchDistributions`. In this manner, algorithms like `PPO` can simply call `MultiTorchDistribution.sample()` without worrying about the specifics of the action space or distributions.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by smorad** [comment]
@sven1977 @kouroshHakha is this still needed, or have you implemented pi heads already?

**Comment by kouroshHakha** [comment]
@smorad This is pulled into an iteration cycle that @ArturNiederfahrenhorst is leading. The PR itself is not needed directly but some parts of it are incorporated in the design.

**Comment by kouroshHakha** [comment]
Let's keep it open for now.
